os/fs/driver/mtd/smart.c

```diff
@@ -3998,7 +3892,7 @@ static int smart_write_alloc_sector(FAR struct smart_struct_s *dev, uint16_t log
  *
  ****************************************************************************/
 
-#ifdef NXFUSE_HOST_BUILD
+#ifdef CONFIG_MTD_SMART_ENABLE_CRC
 static int smart_validate_crc(FAR struct smart_struct_s *dev)
```

Comment:
do you really want to remove NXFUSE_HOST_BUILD? If not sure, use CONFIG_MTD_SMART_ENABLE_CRC as an OR condition

Reply:
Sector header and contents will not be checked by smart_scan function.
smart_validate_crc() NXFUSE host build checks contents of sector while making smartfs image.
Since this task will no longer be done here, this is not required.